pkg/resource/graph/dependency_graph.go

```diff
@@ -123,6 +123,67 @@ func (dg *DependencyGraph) DependenciesOf(res *resource.State) ResourceSet {
 	return set
 }
 
+// `TransitiveDependenciesOf` calculates the set of resources that `r` depends
```

Comment:
It's awesome that this model passes, consider checking it into the "rapid" file: 

```
func TestRapidTransitiveDependenciesOf(t *testing.T) {
	graphCheck(t, func(t *rapid.T, universe []*resource.State) {
		expectedInTDepsOf := transitively(universe)(expectedDependenciesOf)
		dg := NewDependencyGraph(universe)
		for _, a := range universe {
			tda := dg.TransitiveDependenciesOf(a)
			for _, b := range universe {
				assert.Equalf(t,
					expectedInTDepsOf(a, b),
					tda[b],
					"Mismatch on a=%v, b=%b",
					a.URN,
					b.URN)
			}
		}
	})
}
```

The rapid file does not yet test providers so if there's any subtleties with providers won't get caught yet but it will afterward when we extend the random generator.